dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/AbstractTrackerDtoValidationHook.java

```diff
@@ -127,11 +124,20 @@ public void validateTrackedEntity( ValidationErrorReporter reporter, TrackedEnti
         T optionalObject, String value )
     {
         Optional.ofNullable( optionalObject.getOptionSet() )
-            .ifPresent( optionSet -> addErrorIf( () -> optionSet.getOptions().stream().filter( Objects::nonNull )
-                .noneMatch( o -> o.getCode().equalsIgnoreCase( value ) ), reporter, dto, E1125, value,
-                optionalObject.getUid(), optionalObject.getClass().getSimpleName(),
-                optionalObject.getOptionSet().getOptions().stream().filter( Objects::nonNull ).map( Option::getCode )
-                    .collect( Collectors.joining( "," ) ) ) );
+            .ifPresent( optionSet -> reporter.addErrorIf(
+                () -> optionSet.getOptions().stream().filter( Objects::nonNull )
+                    .noneMatch( o -> o.getCode().equalsIgnoreCase( value ) ),
+                () -> TrackerErrorReport.builder()
+                    .uid( dto.getUid() )
+                    .trackerType( dto.getTrackerType() )
+                    .errorCode( E1125 )
+                    .addArg( value )
+                    .addArg( optionalObject.getUid() )
+                    .addArg( optionalObject.getClass().getSimpleName() )
```

Comment:
Looks like a step in the wrong direction if adding multiple args now needs multiple lines.